Change to projecthook calls to use kwargs allowing for newer params

owmtia opened the pull request:
I particular needed access to enable_ssl_verification param but changing the code to use kwargs seemed to be more consistent with other sections of code.

owmtia committed:
Change to projecthook calls to use kwargs allowing for newer params to be used.

FredrikWendt commented on gitlab/__init__.py:
```diff
@@ -586,27 +586,25 @@ def getprojecthook(self, project_id, hook_id):
-        data['merge_requests_events'] = int(bool(merge_requests))
-        data['tag_push_events'] = int(bool(tag_push))
+        if kwargs:
+            data.update(kwargs)
```
Doesn't this change the data sent over the wire from "0/1" (int) to "False/True"?

owmtia committed:
Added loop over kwargs converting to "0/1" (int)

owmtia committed:
Restored previous params, To restore backward compatibility to this function.

owmtia committed:
Documentation updates for projecthook functions

owmtia commented:
I have added the code to change the values to 0/1, I have assumed everything is a bool which seems to be case at the moment with the api. I also restored the previous param as I released it would break backwards compatibility especially as the params have different names to the gitlab API params. I also added the extra documentation to the function.

FredrikWendt commented:
Great.
My only concern is this: if a user of this python lib wants to specify a non-boolean typed argument that is visible and documented in the GitLab API, would they know that the value would be converted to 0 or 1. 

``` python
>>> int(bool("0"))
1
>>> int(bool("False"))
1
>>> int(bool("True"))
1
>>> int(bool("true"))
1
>>> int(bool(True))
1
>>> int(bool(False))
0
>>> int(bool("AnyString"))
1
```

I'm not sure we need to care right now.

owmtia commented:
Yes, I had thought about putting in an isinstance to check for bools/ints but it didn't seem necessary with the current api being all bools.

ehelms commented:
:+1:

owmtia committed:
Fixes for the projecthook documentation so it builds correctly now.

drzraf commented:
1. "note","job","pipeline","wiki_page" and "confidential_issue" are missing
2. "token" is missing and would not be a boolean argument but rather a string.

See: https://docs.gitlab.com/ce/api/projects.html#list-project-hooks

+1 for improving this API endpoint
